### PR TITLE
fixed a bug when button's parent is hided, the button still can responce to clicked event

### PR DESCRIPTION
**Pull request by csy1988525**
fixed a bug when button's parent is hided, the button still can responce to clicked event

### CODE REVIEW

File: extensions/GUI/CCControlExtension/CCControlButton.cpp [hunk 1]
```diff
@@ -638,6 +638,14 @@ bool CCControlButton::ccTouchBegan(CCTouch *pTouch, CCEvent *pEvent)
         return false;
     }
     
+    for (CCNode *c = this->m_pParent; c != NULL; c = c->getParent())
```
Comment (harikrishnar) [inline extensions/GUI/CCControlExtension/CCControlButton.cpp:641]: cant we use the hasVisibleParents() method present in CCControl class?

Comment (csy1988525) [inline extensions/GUI/CCControlExtension/CCControlButton.cpp:641]: yes you can. just do it

File: extensions/GUI/CCControlExtension/CCControlButton.cpp [hunk 2]
```diff
@@ -638,6 +638,14 @@ bool CCControlButton::ccTouchBegan(CCTouch *pTouch, CCEvent *pEvent)
...
+    {
+        if (c->isVisible() == false)
```
Comment (dumganhar) [inline extensions/GUI/CCControlExtension/CCControlButton.cpp:643]: It's better to change it to `if (!c->isVisible())`

Comment (harikrishnar) [inline extensions/GUI/CCControlExtension/CCControlButton.cpp:643]: this is already fixed now, in a previous request

**Pull request merged**